packages/cli/src/ui/hooks/useExtensionUpdates.ts

```diff
@@ -23,13 +23,18 @@ export const useExtensionUpdates = (
   const [extensionsUpdateState, setExtensionsUpdateState] = useState(
     new Map<string, ExtensionUpdateState>(),
   );
-  useMemo(() => {
-    const checkUpdates = async () => {
+  const [isChecking, setIsChecking] = useState(false);
+
+  (async () => {
+    if (isChecking) return;
+    setIsChecking(true);
```

Comment:
be very careful adding logic like this that unconditionally updates react state. This code causes us to render frames continuously causing flicker because we are continuously setting setIsChecking to true and then back to false.